backends/jms/src/main/java/org/hibernate/search/backend/impl/jms/AbstractJMSHibernateSearchController.java

```diff
@@ -69,8 +70,16 @@ public void onMessage(Message message) {
 				log.messageReceivedForUndefinedIndex( indexName );
 				return;
 			}
-			queue = indexManager.getSerializer().toLuceneWorks( (byte[]) objectMessage.getObject() );
-			indexManager.performOperations( queue, null );
+
+			LuceneWorkSerializer luceneWorkSerializer = integrator.getServiceManager().requestService( LuceneWorkSerializer.class );
```

Comment:
You could use it to keep e.g. the JMS implementations untouched.

Your call, I'm just preferring smaller patches and have a bit of concern about the performance of those serializer lookups.